front/src/common/BootstrapSNCF/MultiSelectSNCF.tsx

```diff
@@ -0,0 +1,148 @@
+          value={selectOption.label}
+        >
+          {selectOption.label}
+        </option>
+      )),
+    [selectOptions]
+  );
+
+  const renderSelectToggles = (options: { label: string }[]) =>
+    options.map((toggle, index) => {
+      const { label } = toggle;
+      const isChecked = selectedValues.includes(label);
+      const onClickToggle = () => {
+        onChange(
+          isChecked
+            ? [...selectedValues.filter((selectValue) => selectValue !== label)]
+            : (prevValues) => [...prevValues, label]
+        );
+      };
+      return (
+        <div className="select-menu-item" role="listitem" key={`selectToggle-${label}`}>
+          <div className="custom-control custom-checkbox">
+            <button
+              type="button"
+              data-role="value"
+              data-target={index}
+              role="checkbox"
+              aria-checked="false"
+              className={`custom-control-label w-100 text-left font-weight-medium ${
+                isChecked ? 'active' : ''
+              }`}
+              onClick={onClickToggle}
+            >
+              {label}
+            </button>
+          </div>
+        </div>
+      );
+    });
+
+  const memoizedAllOptionsLabel = useMemo(
+    () => selectOptions.map((option) => option.label),
+    [selectOptions]
+  );
+
+  useEffect(() => {
+    const isAllSelected = memoizedAllOptionsLabel.every((label) => selectedValues.includes(label));
+    if (isAllSelected && !selectAll) setSelectAll(true);
+    if (!isAllSelected && selectAll) setSelectAll(false);
+  }, [selectedValues]);
+
+  useEffect(() => {
+    const isAllSelected = memoizedAllOptionsLabel.every((label) => selectedValues.includes(label));
+    const selects = isAllSelected ? [] : selectedValues;
+    onChange(selectAll ? [...memoizedAllOptionsLabel] : selects);
+  }, [selectAll]);
+
+  return (
+    <>
+      <label htmlFor={`multiSelect-${multiSelectTitle}`}>{multiSelectTitle}</label>
+      <div
+        className={`select-improved ${multiSelectToggle && 'active'}`}
+        data-component="select-multiple"
+      >
+        <div className="select-control">
+          <div className="input-group" data-role="select-toggle">
+            <div className="form-control">
+              {isSelectAll ? (
+                <div className="custom-control custom-checkbox">
+                  <button
+                    type="button"
+                    data-role="value"
+                    role="checkbox"
+                    aria-checked="false"
+                    className={`custom-control-label font-weight-medium ${selectAll && 'active'}`}
+                    onClick={() => setSelectAll((prevState) => !prevState)}
+                  >
+                    {multiSelectSubTitle || multiSelectTitle}
+                  </button>
+                </div>
+              ) : (
+                multiSelectSubTitle || multiSelectTitle
```

Comment:
```suggestion
                selectedValues
```
You can use the function reduce or join to display the selectedValues properly